cmd/eksctl-anywhere/cmd/validatecreatecluster.go

```diff
@@ -0,0 +1,140 @@
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/kubeconfig"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/validations"
+	"github.com/aws/eks-anywhere/pkg/validations/commandvalidations"
+	"github.com/aws/eks-anywhere/pkg/validations/createvalidations"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+type validateOptions struct {
+	clusterOptions
+	hardwareCSVPath       string
+	tinkerbellBootstrapIP string
+}
+
+var valOpt = &validateOptions{}
+
+var validateCreateCmd = &cobra.Command{
+	Use:          "create -f <cluster-config-file> [flags]",
+	Short:        "Validate create cluster",
+	Long:         "This command is used to validate the create cluster action",
+	PreRunE:      bindFlagsToViper,
+	SilenceUsage: true,
+	RunE:         valOpt.validateCreateCluster,
+}
+
+func init() {
+	validateCmd.AddCommand(validateCreateCmd)
+	validateCreateCmd.Flags().StringVarP(&valOpt.fileName, "filename", "f", "", "Filename that contains EKS-A cluster configuration")
+	validateCreateCmd.Flags().StringVarP(
+		&valOpt.hardwareCSVPath,
+		TinkerbellHardwareCSVFlagName,
+		TinkerbellHardwareCSVFlagAlias,
+		"",
+		TinkerbellHardwareCSVFlagDescription,
+	)
+	validateCmd.Flags().StringVar(&valOpt.tinkerbellBootstrapIP, "tinkerbell-bootstrap-ip", "", "Override the local tinkerbell IP in the bootstrap cluster")
+
+	if err := validateCreateCmd.MarkFlagRequired("filename"); err != nil {
+		log.Fatalf("Error marking flag as required: %v", err)
+	}
+}
+
+func (valOpt *validateOptions) validateCreateCluster(cmd *cobra.Command, _ []string) error {
+	ctx := cmd.Context()
+
+	clusterSpec, err := cluster.NewSpecFromClusterConfig(valOpt.fileName, version.Get())
+	if err != nil {
+		return err
+	}
+
+	if clusterSpec.Config.Cluster.Spec.DatacenterRef.Kind == v1alpha1.TinkerbellDatacenterKind {
+		flag := cmd.Flags().Lookup(TinkerbellHardwareCSVFlagName)
+
+		// If no flag was returned there is a developer error as the flag has been removed
+		// from the program rendering it invalid.
+		if flag == nil {
+			panic("'hardwarefile' flag not configured")
+		}
+
+		if !viper.IsSet(TinkerbellHardwareCSVFlagName) || viper.GetString(TinkerbellHardwareCSVFlagName) == "" {
+			return fmt.Errorf("required flag \"%v\" not set", TinkerbellHardwareCSVFlagName)
+		}
+
+		if !validations.FileExists(valOpt.hardwareCSVPath) {
+			return fmt.Errorf("hardware config file %s does not exist", valOpt.hardwareCSVPath)
+		}
+	}
+
+	cliConfig := buildCliConfig(clusterSpec)
+	dirs, err := valOpt.directoriesToMount(clusterSpec, cliConfig)
+	if err != nil {
+		return err
+	}
+
+	tmpPath, err := os.MkdirTemp("./", "tmpValidate")
+	if err != nil {
+		return err
+	}
+	deps, err := dependencies.ForSpec(ctx, clusterSpec).WithExecutableMountDirs(dirs...).
+		WithWriterFolder(tmpPath).
```

Comment:
Can we just use `.WithWriter()` here? It has a built-in temp folder

Reply:
From what I found `WithWriter()` makes the temp folder under the `factory.writerFolder` location, which is set to the cluster name in the spec by the `dependencies.ForSpec` func.

I used the temp path to avoid accidentally cleaning up any existing directory named with the cluster name. Is there a way to avoid using the cluster name directory with `WithWriter()`?